docs/DeveloperGuide.md

```diff
@@ -377,9 +402,262 @@ Use case ends.
 
 3a. The given index is invalid
 
-> 3a1. AddressBook shows an error message <br>
+> 3a1. TaskManager shows an error message <br>
   Use case resumes at step 2
 
+#### Use case: UC02 - Add new task with a task description
+
+**MSS**
+
+1. User enter the new task to be added with all the details including date/time, tags, and description
+2. TaskManager accepts the new task with a confirmation
+Use case ends.
+
+**Extensions**
+
+1a. The input is not valid
+
+> The TaskManager ask for the correct input <br>
+  Use case resumes at step 1
+
+2a. User requests to undo the action
+
+> 2a1. TaskManager shows that undo is successful <br>
+  Use case ends
+
+#### Use case: UC03 - Edit task description
```

Comment:
while this is a very well written use case (:+1:), it gives me an impression that the edit command is a multi step command.
please remember that the commands need to be "one shot" commands.